core/update.php

```diff
@@ -140,27 +142,42 @@ function update_script_selection_form($form, &$form_state) {
     backdrop_flush_all_caches();
   }
   else {
-    $form['help'] = array(
-      '#type' => 'help',
-      '#markup' => 'Updates have been found that need to be applied. You may review the updates below before executing them.',
-      '#weight' => -5,
-    );
-    if ($incompatible_count) {
+    $number_applied = ($count) ? $count - $incompatible_count : 0;
+    if ($number_applied && $incompatible_count) {
       $form['start']['#title'] = format_plural(
         $count,
         '1 pending update (@number_applied to be applied, @number_incompatible skipped)',
         '@count pending updates (@number_applied to be applied, @number_incompatible skipped)',
-        array('@number_applied' => $count - $incompatible_count, '@number_incompatible' => $incompatible_count)
+        array('@number_applied' => $number_applied, '@number_incompatible' => $incompatible_count)
+      );
+      $form['start']['#collapsed'] = FALSE;
+    }
+    elseif ($incompatible_count) {
+      $form['start']['#title'] = format_plural(
+        $count,
+        '1 pending update (@number_incompatible skipped)',
+        '@count pending updates (@number_incompatible skipped)',
+        array('@number_incompatible' => $incompatible_count)
       );
+      $form['start']['#collapsed'] = FALSE;
     }
     else {
+      $form['help'] = array(
+        '#type' => 'help',
+        '#markup' => 'Updates have been found that need to be applied. You may review these updates below.',
+        '#weight' => -5,
+      );
       $form['start']['#title'] = format_plural($count, '1 pending update', '@count pending updates');
     }
+
     $form['actions'] = array('#type' => 'actions');
-    $form['actions']['submit'] = array(
-      '#type' => 'submit',
-      '#value' => t('Apply pending updates'),
-    );
+    // Only add Apply button if there are updates.
+    if ($count) {
+      $form['actions']['submit'] = array(
+        '#type' => 'submit',
+        '#value' => t('Apply pending updates'),
+      );
```

Comment:
[nitpicking] condition could be smaller: `#access => $count`

Reply:
clever!